Assignment 1/test/SimpleRegistryTest.t.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "src/SimpleRegistry.sol";
+
+/// @title Tests the simple name registry.
+/// @author Thomas den Hollander
+/// @notice Used to test the required functionality for the `SimpleRegistry` contract.
+contract SimpleRegistryTest is Test {
+    SimpleRegistry simpleRegistry;
+
+    constructor() {
+        simpleRegistry = new SimpleRegistry();
+    }
+
+    /// @notice Test claiming some names for this contract.
+    function testClaimNames() public {
+        assertEq(simpleRegistry.owners("TestName"), address(0));
+        simpleRegistry.claimName("TestName");
+        assertEq(simpleRegistry.owners("TestName"), address(this));
+        
+        assertEq(simpleRegistry.owners("A-Somewhat_longer_test!Name"), address(0));
+        simpleRegistry.claimName("A-Somewhat_longer_test!Name");
+        assertEq(simpleRegistry.owners("A-Somewhat_longer_test!Name"), address(this));
+    }
+
+    /// @notice Test releasing a name for this contract.
+    /// @param name The name to claim and release.
+    function testClaimAndRelease(string calldata name) public {
```

Comment:
At this point you should build a new state (NameClaimed) so that you can avoid claiming a name in all your subsequent tests. Check this from [James](https://sabnock.substack.com/p/take-your-solidity-testing-to-the).